src/components/CommentCard.jsx

```diff
@@ -1,10 +1,55 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteCommentById } from "../api";
 
-export default function CommentCard({ comment }) {
+export default function CommentCard({ comment, setComments }) {
   const parsedDate = Date.parse(comment.created_at);
+
   const formattedDate = new Date(parsedDate).toLocaleString("en-GB", {
     timeZone: "UTC",
   });
+
+  const { loggedInUser, setLoggedInUser } = useContext(UserContext);
+  const [deleteBtn, setDeleteBtn] = useState(false);
+  const [deleteComment, setDeleteComment] = useState({});
+
+  useEffect(() => {
+    loggedInUser.username === comment.author
+      ? setDeleteBtn(true)
+      : setDeleteBtn(false);
+  }, [loggedInUser]);
+
+  useEffect(() => {
+    if (deleteComment.comment_id) {
+      setTimeout(
+        () =>
+          setComments((currCommets) => {
+            return [...currCommets];
+          }),
+        100
+      );
+      //has to be existing post delete
+      console.log(deleteComment.comment_id);
+      deleteCommentById(deleteComment.comment_id)
+        .then((res) => {
+          //204 only after a
+          console.log(res);
+          alert("deleted successfully");
```

Comment:
Alerts are a bit notorious for being a bad user experience, it would be much better if we could get this just being shown on the page instead of being an annoying alert